server/src/main/java/org/elasticsearch/search/rank/feature/CustomRankInput.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.search.rank.feature;
+
+import org.elasticsearch.common.io.stream.Writeable;
+
+/**
+ * Defines custom rank input that we send in as input to rank retrievers. Placeholder for future customizations outside snippets.
+ */
+public interface CustomRankInput extends Writeable {}
```

Comment:
I don't think this is needed, at least not in this form. 
Instead we could expose a `RankFeaturePhaseRankShardContext#prepareForFetch(SearchContext context)` that would be overridable by the `RerankingRankFeaturePhaseRankShardContext`?